x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/JobManagerTests.java

```diff
@@ -186,6 +186,7 @@ public void onFailure(Exception e) {
 
     // Check that for new jobs, a default value is set for the model_prune_window field,
     // if one is not present in the configuration.
+    @SuppressWarnings("unchecked")
```

Comment:
Is it just `ActionListener<Boolean> listener = (ActionListener<Boolean>) invocation.getArguments()[2]` that needs this?  If so, you should be able to put the suppression immediately before that line instead of applying to the whole method.

Reply:
No, there's a few others in that test method as well
```
/Users/eds/src/elasticsearch/es-new/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/JobManagerTests.java:206: warning: [unchecked] unchecked method invocation: method createJobResultIndex in class JobResultsProvider is applied to given types
        }).when(jobResultsProvider).createJobResultIndex(requestCaptor.capture(), any(ClusterState.class), any(ActionListener.class));
                                                        ^
  required: Job,ClusterState,ActionListener<Boolean>
  found:    Job,ClusterState,ActionListener
/Users/eds/src/elasticsearch/es-new/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/JobManagerTests.java:206: warning: [unchecked] unchecked conversion
        }).when(jobResultsProvider).createJobResultIndex(requestCaptor.capture(), any(ClusterState.class), any(ActionListener.class));
                                                                                                              ^
  required: ActionListener<Boolean>
  found:    ActionListener
error: warnings found and -Werror specified
1 error
2 warnings
```
Given this I think it's cleaner to apply the annotation to the whole method.